src/nvim/api/private/defs.h

```diff
@@ -36,7 +36,7 @@ typedef enum {
 } MessageType;
 
 /// Used as the message ID of notifications.
-#define NO_RESPONSE UINT64_MAX
+#define NO_RESPONSE UINT32_MAX
```

Comment:
Maybe this will be an issue, as `UINT32_MAX` is practially reachable unlike `UINT64_MAX`. Perhaps we should still use `uint64_t` internally, and explicitly implement wrap-around, so transcendentals can be used to distinguish sentinel values like this one.

Reply:
Yes, I thought about it. It seems that the only thing to do is conversion to wire type. But how to convert? Take message id modulo to `UINT32_MAX`? Well, this means that we still use `uint32_t`. Also, I think that long inner representation could result in some non-trivial issues if one tries to perform arithmetic and logic operations on numbers longer than 32-bits.

Actully, goal of message id is to link request message and response message. Common policy of picking a message id is incrementing of per connection counter. So one just could take a random interger from `[0, UINT32_MAX]`. Since C/C++ standard declares that overflow for unsigneds is correct behaviour, it is absolutely the same as `uint64_t` modulo to `UINT32_MAX` but a bit safer. In any case the length effective window of ids is `4294967296`.

Reply:
I see. I think that it is better to rely on complier and its type system than mannually handle type conversion in advance.

@justinmk What do you think?

Reply:
@daskol It's easier to see that something is wrong if the value is `>UINT32_MAX` (internally)--that means we didn't wrap correctly. Whereas with implicit wrap, we won't notice an accidental/unwanted wrap-around. How will the compiler/type system will improve anything here?

Reply:
Resolved in https://github.com/neovim/neovim/pull/9887 by elminated NO_RESPONSE sentinel value.